### PR TITLE
fix(otlp): replace otlp trace attr type from string to jsonb

### PR DESCRIPTION
**Pull request by paomian**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)

https://github.com/GreptimeTeam/greptimedb/issues/4873

## What's changed and what's your intention?

* replace otlp trace attr type from string to jsonb
* change default trace table name and specify the trace table name via the header

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

**Review by sunng87 — APPROVED** [review]

**Comment by sunng87** [comment]
we need to add a test case for trace table name customization

**Review by shuiyisong — APPROVED** [review]
Rest LGTM

**Comment by shuiyisong** [comment]
Also is this a breaking change?

**Comment by paomian** [comment]
> Also is this a breaking change?

Currently trace is not open to the public. So these changes are fine.

File: src/servers/src/otlp/trace.rs [hunk 1]
```diff
@@ -23,7 +23,7 @@ use crate::error::Result;
 use crate::row_writer::{self, MultiTableData, TableData};
 
 const APPROXIMATE_COLUMN_COUNT: usize = 24;
```
[the remark this answers is not in the document]
Comment (paomian) [reply src/servers/src/otlp/trace.rs:25]: Here is just a speculative number of columns, unnecessarily identical.

**Pull request merged**